data-prepper-core/src/test/java/org/opensearch/dataprepper/peerforwarder/client/PeerForwarderClientTest.java

```diff
@@ -58,27 +68,42 @@ class PeerForwarderClientTest {
     private static final String TEST_PIPELINE_NAME = "test_pipeline_name";
 
     private ObjectMapper objectMapper;
+    @Mock
+    private PluginMetrics pluginMetrics;
+
+    @Mock
+    private PeerForwarderConfiguration peerForwarderConfiguration;
 
     @Mock
-    PeerForwarderConfiguration peerForwarderConfiguration;
+    private PeerClientPool peerClientPool;
 
     @Mock
-    PeerClientPool peerClientPool;
+    private PeerForwarderClientFactory peerForwarderClientFactory;
 
     @Mock
-    PeerForwarderClientFactory peerForwarderClientFactory;
+    private Counter requestsCounter;
+
+    private NoopTimer timer;
 
     @BeforeEach
     void setUp() {
+        timer = new NoopTimer(new Meter.Id("test", Tags.empty(), null, null, Meter.Type.TIMER));
```

Comment:
You could make this a `spy` and then verify that it is actually called. The spy will allow you to continue to call real methods (and thus get the behavior you need to call the lambda). And you can verify behavior.

Reply:
I don't consider this a must for merging though.

Reply:
I tried it but unfortunately we can't create spy on `NoopTimer`.